cyclonedx_py/_internal/utils/pep639.py

```diff
@@ -38,43 +39,110 @@
     from cyclonedx.model.license import License
 
 
+def handle_bad_license_file_encoding(
+    dist: 'Distribution',
+    lfile: str,
+    logger: 'Logger'
+) -> Union[str, None]:
+
+    def try_load(dist: 'Distribution', metadir: str, filename: str) -> Union[str, None]:
+        # Might raise NotImplementedError in theory
+        # but nothing we can do in that case.
+        try:
+            candidate = dist.locate_file(join(metadir, filename))
+        except NotImplementedError:
+            return None
+
+        if not candidate:
+            return None
+
+        try:
+            with open(str(candidate), 'rb') as fin:
+                return io2str(fin)
+        except FileNotFoundError:
+            pass
+        return None
+
+    # Distribution has no method to find the actual metadata dir,
+    # e.g. dist-info or egg-info.
+    # So we mimic the logic in PathDistribution and check both subdirs
+    content: Union[str, None] = None
+    for metadir in ('.dist-info', '.egg-info'):
+        content = try_load(dist, metadir, lfile)
+        if content:
+            break
+
+    if content is None:
+        logger.debug('Error: license file %r for dist %r is not UTF-8 encoded',
+                     lfile, dist.metadata['Name'])
+    return content
+
+
+def gather_license_texts(
+    dist: 'Distribution',
+    lfiles: Set[str],
+    logger: 'Logger'
+) -> Generator['License', None, None]:
+    lack = LicenseAcknowledgement.DECLARED
+    for mlfile in lfiles:
+        # see spec: https://peps.python.org/pep-0639/#add-license-file-field
+        # latest spec rev: https://discuss.python.org/t/pep-639-round-3-improving-license-clarity-with-better-package-metadata/53020  # noqa: E501
+
+        # per spec > license files are stored in the `.dist-info/licenses/` subdirectory of the produced wheel.
+        # but in practice, other locations are used, too.
+        # loop over the candidate location and pick the first one found.
+        locations = ('licenses', 'license_files', '.')
+        malformed = None
+        content = None
+        for loc in locations:
+            try:
+                path = join(loc, mlfile)
```

Comment:
`path = join()` not needed to be in the try/except.
mote outside

Reply:
Fixed.